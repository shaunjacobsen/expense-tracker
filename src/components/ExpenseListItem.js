import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = (props) => (
  <div>
    <strong>${props.amount / 100}</strong> <Link to={`/edit/${props.id}`}>{props.description}</Link>
  </div>
);

export default ExpenseListItem;