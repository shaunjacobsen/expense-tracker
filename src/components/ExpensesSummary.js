import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from './../selectors/expenses';
import expensesTotal from './../selectors/expenses-total';

export const ExpensesSummary = props => (
  <div>
    Viewing {props.expenseCount} expense{props.expenseCount === 1
      ? ' '
      : 's '}
    totalling {numeral(props.total / 100).format('$0,0.00')}
  </div>
);

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    total: expensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
