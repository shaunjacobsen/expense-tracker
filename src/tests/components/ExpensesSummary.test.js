import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from './../../components/ExpensesSummary';

test('it should render ExpensesSummary properly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} />);
  expect(wrapper).toMatchSnapshot();
});

test('it should render ExpensesSummary properly with several expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={5} />);
  expect(wrapper).toMatchSnapshot();
});
