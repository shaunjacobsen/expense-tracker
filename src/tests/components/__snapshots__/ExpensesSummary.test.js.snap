// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render ExpensesSummary properly with one expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Viewing ",
        1,
        " expense",
        " ",
        "totalling ",
        "$0.00",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Viewing ",
      1,
      " expense",
      " ",
      "totalling ",
      "$0.00",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          1,
          " expense",
          " ",
          "totalling ",
          "$0.00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        1,
        " expense",
        " ",
        "totalling ",
        "$0.00",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`it should render ExpensesSummary properly with several expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Viewing ",
        5,
        " expense",
        "s ",
        "totalling ",
        "$0.00",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Viewing ",
      5,
      " expense",
      "s ",
      "totalling ",
      "$0.00",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          5,
          " expense",
          "s ",
          "totalling ",
          "$0.00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        5,
        " expense",
        "s ",
        "totalling ",
        "$0.00",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
